### ISSUE TITLE
Aria accessibility just lacking

### ISSUE THREAD

**Issue by AndyBean** [selectize/selectize.js#798]
http://www.w3.org/TR/wai-aria-practices/#autocomplete

I got valid pushback from a customer recently wanting to ensure this widget met accessibility standards. It is missing a few Aria tags, and that is it. I may not have time today but hope to find time soon to add a PR that adds the missing components for others. It is keyboard navigable already and I think in a way that meets spec. Just missing are the screenreader aspects.

"The widget has a role of combobox, and its parent role is input. It has a child with role textbox. The property aria-autocomplete indicates whether user input completion suggestions are provided."

**Comment by AndyBean** [comment]
EDIT:  I should add this is not a problem that Brian has to solve, any of us can and should. I am just logging the issue here so I can track it and get to it myself if no one beats me to it.

**Comment by u01jmg3** [comment]
Any movement with this? Would be curious to know what is missing to enable Screen Reader support. Thanks

**Comment by u01jmg3** [comment]
I've semi put together a plugin for this but I would appreciate any guidance.

I'm getting `blank` in [NVDA 2015.3 (Speech Viewer)](http://www.nvaccess.org/download/) when I cycle through Selectize options with my keyboard. Also worth noting I'm using `optgroups`.

``` js
Selectize.define('accessibility', function(options){
    var self = this;

    self.setup = (function(){
        var original = self.setup;

        return function(){
            original.apply(self, arguments);
            self.$wrapper.attr({'role': 'application'});
            self.$dropdown.attr({'id': 'aria-selectize-dropdown', 'role': 'listbox'});
            self.$control_input.attr({
                'aria-owns'        : 'aria-selectize-dropdown',
                'aria-haspopup'    : 'true',
                'role'             : 'combobox',
                'aria-autocomplete': 'list'
            });
        };
    })();
});

$('.selectize').selectize({
    plugins: {'accessibility': {}},
    render: {
        option: function($item, escape){
            return '<div role="option">' + $item.value + '</div>';
        }
    }
});
```

---
- Generated HTML which is not yet defined correctly for screen readers

``` html
<div class="selectize-control selectize multi plugin-accessibility" role="application">
    <div class="selectize-input items not-full has-options">
        <input type="text" autocomplete="off" tabindex="" aria-owns="aria-selectize-dropdown" aria-haspopup="true" role="combobox" aria-autocomplete="list" style="width: 4px; opacity: 1; position: relative; left: 0px;" data-cip-id="cIPJQ342845639">
    </div>
    <div class="selectize-dropdown multi selectize plugin-accessibility" id="aria-selectize-dropdown" role="listbox" style="display: none; width: 945px; top: 34px; left: 0px; visibility: visible;">
        <div class="selectize-dropdown-content">
            <div data-group="programme_titles.title" class="optgroup">
                <div class="optgroup-header">Programme Titles</div>
                <div data-value="programme_titles.title.Mathematics" data-selectable="" role="option">Mathematics</div>
                <div data-value="programme_titles.title.Mathematics and Philosophy" data-selectable="" role="option">Mathematics and Philosophy</div>
                <div data-value="programme_titles.title.Mathematics and Physics" data-selectable="" role="option">Mathematics and Physics</div>
            </div>
        </div>
    </div>
</div>
```

**Comment by u01jmg3** [comment]
@kirstenmalin could you help? I saw your [commit](https://github.com/kirstenmalin/chosen/commit/4d0da63e3839df229c3095e692395acbd39d811a) for Chosen adding accessibility support

**Comment by u01jmg3** [comment]
@binary-koan could you help? I saw your [PR](https://github.com/select2/select2/pull/3821) for Select2 adding accessibility support

**Comment by binary-koan** [comment]
Hi, sorry for the late reply. I think the key thing you're missing is `aria-activedescendant` - this needs to be on the input box (or the control which is focused while you're cycling through options) and should contain the ID of the list item which is currently focused. [The Google autocomplete lesson](https://webaccessibility.withgoogle.com/unit?unit=6&lesson=13) was the biggest help with my select2 PR, so it might be worth a read.

Let me know if it works!

**Comment by u01jmg3** [comment]
Thanks for replying!

Already tried `aria-activedescendant` with little success (NVDA didn't seem to pick it up) but will try again after I've checked out that link

**Comment by joselogil** [comment]
hi @u01jmg3. have you found a way to have this working?

**Comment by u01jmg3** [comment]
Nope - the most I could do was write the plugin I have provided

**Comment by SalmenBejaoui** [comment]
Hi @laurelstreng,
It's a pleasure to read it. You can follow the project here [https://github.com/SLMNBJ/selectize-plugin-a11y](selectize-plugin-a11y). We are developing a second version. More robust, adding custom labes and support native Selectize events.

**Comment by rskaarup** [comment]
Magnificent, thank you for making this plugin @SLMNBJ

**Comment by AssassinLV** [comment]
Bump....
Probably will have to consider other plugins that has no issues - as WCAG 2.1 AA is an EU requirement,
